test/functional/test_cluster_basic.py

```diff
@@ -82,3 +66,55 @@ def test_git_type_demo_project_config(self):
             build_id=build_id, expected_data={'num_atoms': 10, 'num_subjobs': 10})
         self.assert_build_artifact_contents_match_expected(
             build_id=build_id, expected_build_artifact_contents=expected_artifact_contents)
+
+    @skip('This test currently fails due to https://github.com/box/ClusterRunner/issues/296.')
+    def test_slave_reconnection_does_not_take_down_master(self):
+        test_config = JOB_WITH_SETUP_AND_TEARDOWN
+        job_config = yaml.safe_load(test_config.config[os.name])['JobWithSetupAndTeardown']
+        master = self.cluster.start_master()
+
+        # Start a slave, hard kill it, then reconnect it.
+        self.cluster.start_slave(num_executors_per_slave=5, start_port=43001)
+        self.cluster.kill_slaves(kill_gracefully=False)
+
+        # Make sure the slave restarts with the same port.
+        slave = self.cluster.start_slave(num_executors_per_slave=5, start_port=43001)
```

Comment:
So this test, if run on the current codebase, takes down the master? I thought the problem occurred when the slave is running BUILD_A, and then gets a setup request sent from the master for BUILD_B before BUILD_A has completed.